adb_graphics/figures/maps.py

```diff
@@ -300,12 +302,29 @@ def draw(self, show=False):
         # Levels should be included in the settings dict here since they don't
         # correspond to a full field of contours.
         if self.hatch_fields:
+            handles = []
             for field in self.hatch_fields:
-                self._draw_field(ax=ax,
-                                 field=field,
-                                 func=self.map.m.contourf,
-                                 **field.contour_kwargs,
-                                 )
+                colors = field.contour_kwargs.get('colors', 'k')
+                hatches = field.contour_kwargs.get('hatches', '----')
+                labels = field.contour_kwargs.get('labels', 'XXXX')
+                handles.append(mpatches.Patch(edgecolor=colors[-1], facecolor='lightgrey', \
+                               label=labels[-1], hatch=hatches[-1]))
+
+                cf = self._draw_field(ax=ax,
+                                      field=field,
+                                      func=self.map.m.contourf,
+                                      **field.contour_kwargs,
+                                      )
+
+                # For each level, we set the color of its hatch
+                for i, collection in enumerate(cf.collections):
+                    collection.set_edgecolor(colors[i % len(colors)])
```

Comment:
This line is a bit odd. I'm not sure how many collections we are iterating over, but length of colors is 1 (as defined in the `default_specs.yml` entry), so we'd always get `colors[0]`.

If you change the entry in `default_specs.yml` to a string, you could change this line to just:
```
collection.set_edgecolor(colors)
```

That would get a bit more consistent solution between settings.

I will mention that if you set the `default_specs.yml` setting to a string like `green` you'd get something like `1 % 5 = 1` and `colors[1] = 'r'` which will be an unexpected result.